flask_core/app.py

```diff
@@ -39,4 +55,8 @@ def create_app(config=None):
     # Register all our middleware
     app.wsgi_app = Handler(app.wsgi_app)
 
+    # Set up magic flag generator (ty closure)
+    app.all_flag_ids = app.config["FLAG_IDS"].split(",")
+    app.gen_flag = lambda zid, flag_id: gen_flag(app, zid, flag_id)
+    app.check_flag = lambda zid, flag: check_flag(app, zid, flag)
```

Comment:
I forgot, you can use `type.MethodType` here instead.

```python3
app.check_flag = types.MethodType(check_flag, app)
```

Reply:
ok that's awesome god bless